EDM-2865: Upgrade to Patternfly v6

celdrake opened the pull request:
Yet to be done: review if the YAML editor is aligned with newer versions.

celdrake committed:
Modified PF dependencies

celdrake committed:
Both apps build fine

celdrake committed:
Fixing layout etc

celdrake commented on libs/ui-components/src/components/AuthProvider/AuthProviderDetails/RoleAssigmentDetails.tsx:
```diff
@@ -28,15 +28,15 @@ const RoleAssigmentDetails = ({ roleAssignment }: { roleAssignment: AuthRoleAssi
     <>
       <Label color="purple">{getAssignmentTypeLabel(roleAssignment.type, t)}</Label>
-      <LabelGroup className="pf-v5-u-mt-sm">
+      <LabelGroup className="pf-v6-u-ml-sm">
```
it was missing horizontal spacing

celdrake committed:
remove details page card CSS

celdrake committed:
using tabsnav

celdrake committed:
spacing for all pages on top

celdrake committed:
Login Page

celdrake committed:
Final changes

celdrake committed:
Remove redundant Content components

celdrake committed:
Style using own classes

celdrake committed:
Remove unnecessary FlightctlCardBody component

celdrake committed:
Fix console errors related to DonutCharts

celdrake committed:
Fix weird shape of save button with disabled reason

celdrake committed:
Fix duplicated text appearing in expandable sections

celdrake committed:
fix text for collapsed blocks in device template

celdrake committed:
Fix bug in PF6 for nexted ExpandableSection when collapsed

celdrake committed:
Fixing bolder fonts

jgyselov commented on libs/ui-components/src/components/form/validations.ts:
```diff
@@ -596,11 +596,11 @@ export const validateCPULimit = (cpu: string | undefined): boolean => {
-  if (!trimmed) {
-    return true;
+
+  if (!/^[0-9]+$/.test(cpu)) {
```
This change results in the suggested value `0.4` being invalid. Not sure what's correct here but either the validation or the helper text needs to change.

<img width="363" height="176" alt="Image" src="https://github.com/user-attachments/assets/5df0acc3-69cc-4655-aa6d-ac358f8177d1" />

celdrake replied:
Good catch, thanks!

This was so that 3433asf would be considered an invalid value, but I forgot about the fractions. I'll double check that it's enough to add the `dot` to the regular expression.

Could be part of another PR but I guess since I was testing all fields etc I just noticed that.

jgyselov replied:
Up to you :slightly_smiling_face:

celdrake replied:
Fixed now

jgyselov reviewed (changes requested):
Just one potential little problem. 🙂 

As for the PF6 updates, I didn't see any issues in the code or the UI (the parts that I managed to interact with).

celdrake commented:
> Just one potential little problem. 🙂
> 
> As for the PF6 updates, I didn't see any issues in the code or the UI (the parts that I managed to interact with).

Great, thanks! 

The "Devices" area is a bit tricker to test because you need to either enroll a device (it can be done easily in the backend project doing `make agent-vm`, but it's a bit slow), or simulate one (but then you can't have the device change its status, etc).

Let me know if you'd want to proceed further or you're satisfied with what you covered.

jgyselov commented:
> > Just one potential little problem. 🙂
> > As for the PF6 updates, I didn't see any issues in the code or the UI (the parts that I managed to interact with).
> 
> Great, thanks!
> 
> The "Devices" area is a bit tricker to test because you need to either enroll a device (it can be done easily in the backend project doing `make agent-vm`, but it's a bit slow), or simulate one (but then you can't have the device change its status, etc).
> 
> Let me know if you'd want to proceed further or you're satisfied with what you covered.

I managed to simulate a fleet so I could see the tables, the detail pages, and the edit forms etc. It all looked fine and the changes in the code weren't complicated anyway. I think you're good here.

celdrake committed:
Fix regexp to accept valid fractional numbers

celdrake committed:
Adapt LoginCommandModal to PF6

celdrake commented:
@jgyselov I fixed the CPU validation issue, rebased to latest "main" (and adapted the new LoginCommand component to PF6).
IMO this should be good now, could you please take a final look?
